src/vivarium/framework/results/strategies.py

```diff
@@ -188,6 +188,72 @@ def __call__(self, data: GroupBy) -> Result:
         return Result(self._measure, result, self._additional_keys)
 
 
+class Grouper:
+    """Strategy for grouping data for aggregation and then cleaning it up."""
+
+    # These two attributes are used to give results producer
+    # strategies consistent data types to work with.
+    # E.g. if there are no groupby columns, we need a
+    # dummy column in order to get a groupby object to get the same
+    # behavior out of the `.apply` method in aggregation.
+    default_grouper = 'default_grouper'
+    default_grouper_value = True
+
+    def __init__(self, data_filter: str, groupby_cols: List[str]):
+        """
+        Parameters
+        ----------
+        data_filter
+            A filter to apply before grouping to subset the data.
+        groupby_cols
```

Comment:
A filter, expressed as a Pandas query string, to apply before grouping to subset the data.